test/unit/Reflection/ReflectionClassTest.php

```diff
@@ -1500,6 +1502,83 @@ public function testGetTraitAliases(): void
         ], $classInfo->getTraitAliases());
     }
 
+    public function testGetTraitNamesWithMissingTraitDefinitions(): void
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+            $this->astLocator,
+        ));
+
+        $this->expectException(IdentifierNotFound::class);
+
+        $reflector->reflectClass(Fixture\ClassUsesUnknownTrait::class)->getTraitNames();
+    }
+
+    public function testGetTraitsWithMissingTraitDefinitions(): void
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+            $this->astLocator,
+        ));
+
+        $this->expectException(IdentifierNotFound::class);
+
+        $reflector->reflectClass(Fixture\ClassUsesUnknownTrait::class)->getTraits();
+    }
+
+    public function testGetTraitAliasesWithMissingTraitDefinitions(): void
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+            $this->astLocator,
+        ));
+
+        self::assertSame(
+            [],
+            $reflector->reflectClass(Fixture\ClassUsesUnknownTrait::class)->getTraitAliases(),
+        );
+    }
+
+    /** @param list<class-string> $expectedTraits */
+    #[DataProvider('getTraitClassNamesDataProvider')]
+    public function testGetTraitClassNames(string $sourcePath, string $className, array $expectedTraits): void
+    {
+        $reflector = new DefaultReflector(new SingleFileSourceLocator(
+            $sourcePath,
+            $this->astLocator,
+        ));
+
+        self::assertSame(
+            $expectedTraits,
+            $reflector->reflectClass($className)->getTraitClassNames(),
+        );
+    }
+
+    /** @return list<array{string, class-string, list<class-string>}> */
+    public static function getTraitClassNamesDataProvider(): array
+    {
+        return [
+            [
+                __DIR__ . '/../Fixture/ClassUsesUnknownTrait.php',
+                Fixture\ClassUsesUnknownTrait::class,
+                ['Roave\BetterReflectionTest\Fixture\UnknownTrait'],
+            ],
```

Comment:
this first test-case shows that we are now able to extract the trait names even if not autoloadable.

the remaining cases add coverage for a few more related cases